src/json_deser.c

```diff
@@ -401,3 +414,140 @@ corto_int16 json_deserialize(corto_object o, corto_string s) {
     }
     return -1;
 }
+
+/*
+ * This method changes the id!
+ */
+static void json_splitParentAndName(char* id, char** parentName, char** name)
+{
+    char *ptr = strrchr(id, '/');
+    if (ptr) {
+        *name = ptr + 1;
+        *parentName = id;
+        *ptr = '\0';
+    } else {
+        *name = id;
+        *parentName = NULL;
+    }
+}
+
+static corto_object json_declareObjectByParentNameNameTypeName(const char* parentName, const char* name, const char* typeName)
+{
+    corto_object package = NULL;
+    corto_object type = corto_resolve(NULL, (corto_string)typeName);
+    if (!type) {
+        corto_seterr("Cannot find %s", typeName);
+        goto errorTypeNotFound;
+    }
+    if (!corto_instanceof(corto_type_o, type)) {
+        corto_seterr("%s is not a type", typeName);
+        goto errorNotType;
+    }
+    if (parentName) {
+        corto_object parent = corto_resolve(root_o, (corto_string)parentName);
+        if (!parent) {
+            corto_seterr("Cannot find parent package %s", parentName);
+            goto errorNoParent;
+        }
+        package = corto_declareChild(parent, (corto_string)name, type);
+        corto_release(parent);
+    } else {
+        package = corto_declareChild(root_o, (corto_string)name, type);
+    }
+    corto_release(type);
+
+    return package;
+
+errorNoParent:
+errorNotType:
+    corto_release(type);
+errorTypeNotFound:
+    return NULL;
+}
+
+static corto_object json_declareObjectByIdAndType(const char* id, const char* type)
+{
+    char* idbuf = corto_strdup(id);
+    if (!idbuf) {
+        goto errorIdBuf;
+    }
+    char *name = NULL;
+    char *parentName = NULL;
+    json_splitParentAndName(idbuf, &parentName, &name);
+
+    corto_object o = json_declareObjectByParentNameNameTypeName(parentName, name, type);
+    corto_dealloc(idbuf);
+    return o;
+
+errorIdBuf:
+    return NULL;
+}
+
+corto_int16 json_toObject(corto_object* o, corto_string s)
+{
+    JSON_Value* topValue = json_parse_string(s);
+    if (!topValue) {
+        corto_seterr("Error parsing %s", s);
```

Comment:
error strings are never capitalized. also, it is good practice to put the library that generated the error in front of the error: `corto_seterr("json: error parsing '%s'", s);`
Also put the string between '' (not "") so that if the string is empty, it will be easier to see.